src/network/taraxa_capability.cpp

```diff
@@ -1148,24 +1150,26 @@ void TaraxaCapability::sendTransactions() {
   }
 }
 
-void TaraxaCapability::doBackgroundWork() {
+void TaraxaCapability::checkLiveness() {
   auto host = host_.lock();
   if (!host) {
     return;
   }
-  for (auto const &peer : peers_) {
-    // Disconnect any node that did not send any message for 3 status intervals
-    if (!peer.second->checkStatus(5)) {
-      LOG(log_nf_) << "Host disconnected, no status message in " << 5 * check_status_interval_ << " ms" << peer.first;
-      host->disconnect(peer.first, p2p::PingTimeout);
-    }
-    // Send status message
-    else {
-      sendStatus(peer.first, false);
+  {
+    boost::shared_lock<boost::shared_mutex> lock(peers_mutex_);
+    for (auto const &peer : peers_) {
+      // Disconnect any node that did not send any message for 3 status intervals
+      if (!peer.second->isAlive(5)) {
```

Comment:
maybe this 5(max_check_count) should be config value? or at least class const?

Reply:
I agree, this part is just an old code that was there before